### PR TITLE
Documentation: add deployment scripts and Jenkinsfile

### PR DESCRIPTION
**Pull request by neubs-bsi**
Signed-off-by: Stephanie Neubauer <Stephanie.Neubauer@bosch-si.com>

#### Type of Change
- [ ] Bug fix
- [ ] New feature
- [ ] Improvements
- [ ] Documentation update
- [ ] Other

#### How Has This Been Tested?
- [ ] Specific Unit Test
- [ ] Executing the Example Projects
- [ ] `mvn test` of whole project
- [ ] `mvn test` of submodule
- [ ] `gradle test` (for the gradle plugin)

#### Checklist
- [ ] My code follows the style and guidelines of this project
- [ ] I have self-reviewed my code 
- [ ] I have provided tests that prove my change is working 
- [ ] Existing tests still pass with my changes 
- [ ] I have updated the documentation accordingly to my changes

### CODE REVIEW

File: antenna-documentation/scripts/createOrUpdateVersionDocumentationFolder.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#
+# SPDX-License-Identifier: EPL-2.0
+BASEDIR=$(readlink -f "$0")
+BASEDIR=$(dirname "$BASEDIR")
+
+set -e
+
+CURRENT_BRANCH=$1
+# checkout branch and build antenna
+git checkout ${CURRENT_BRANCH}
```
Comment (sschuberth) [inline antenna-documentation/scripts/createOrUpdateVersionDocumentationFolder.sh]: Why is a `git checkout` necessary at all? If we have this file, we already have a Git working tree. Isn't it good enough to then be able to run this script to create docs form the current branch? Or do we need to be able to create docs for other branches, too, and these other branches also don't have this script yet?

File: antenna-documentation/scripts/createOrUpdateVersionDocumentationFolder.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+mvn clean install -DskipTests
+mvn site -f antenna-documentation/pom.xml
+
+cp -R antenna-documentation/target/documentation/ temp_documentation/
+
+git checkout gh-pages
+
+rsync -av temp_documentation/ ${CURRENT_BRANCH}/
```
Comment (sschuberth) [inline antenna-documentation/scripts/createOrUpdateVersionDocumentationFolder.sh]: Why first copy to a temporary location? The `git checkout gh-pages` shouldn't tamper with the untracked `antenna-documentation/target/documentation/` directory, so we should be able to directly copy from there. Also, why once use `cp` and once `rsync` to copy directories? I'd prefer to align on either one for consistency.

File: antenna-documentation/scripts/createOrUpdateVersionDocumentationFolder.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+BASEDIR=$(readlink -f "$0")
+BASEDIR=$(dirname "$BASEDIR")
+
+set -e
```
Comment (sschuberth) [inline antenna-documentation/scripts/createOrUpdateVersionDocumentationFolder.sh]: You could include this into the shebang line, like `#!/usr/bin/env bash -e`.

File: antenna-documentation/scripts/documentation_release.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+BASEDIR=$(readlink -f "$0")
+BASEDIR=$(dirname "$BASEDIR")
+
+set -e
```
Comment (sschuberth) [inline antenna-documentation/scripts/documentation_release.sh]: You could include this into the shebang line, like `#!/usr/bin/env bash -e`.

File: antenna-documentation/scripts/documentation_release.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+CURRENT_BRANCH=${git describe --tag}
+
+createReleaseDocumentation() {
+  chmod a+x createOrUpdateVersionDocumentationFolder.sh
+  sh ./createOrUpdateVersionDocumentationFolder.sh ${CURRENT_BRANCH}
```
Comment (sschuberth) [inline antenna-documentation/scripts/documentation_release.sh]: The `sh` prefix here overrides the shebang line in the script. So on systems where `sh` is not an alias for `bash`, you don't get Bash, but we want Bash, so you should omit the prefix.

File: antenna-documentation/scripts/documentation_release.sh [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+# SPDX-License-Identifier: EPL-2.0
+BASEDIR=$(readlink -f "$0")
+BASEDIR=$(dirname "$BASEDIR")
+
+set -e
+
+CURRENT_BRANCH=${git describe --tag}
+
+createReleaseDocumentation() {
+  chmod a+x createOrUpdateVersionDocumentationFolder.sh
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply antenna-documentation/scripts/documentation_release.sh]: Similar below.

File: antenna-documentation/scripts/documentation_release.sh (hunk 5 above)
Comment (sschuberth) [reply antenna-documentation/scripts/documentation_release.sh]: Similar below.

File: antenna-documentation/scripts/documentation_release.sh [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  git checkout ${CURRENT_BRANCH}
+}
+
+createNewIndexRootFile() {
+  chmod a+x writeIndexRootFile.sh
+  sh ./writeIndexRootFile.sh
+}
+
+pushChanges() {
+  git push `git config remote.origin.url | sed "s/https:\/\//https:\/\/${GITHUB_USER}:${GITHUB_PASSWORD}@/g"` gh-pages
```
Comment (sschuberth) [inline antenna-documentation/scripts/documentation_release.sh]: Sometimes you use `$()` and sometimes backticks for command expansion. Aligning on either one would be good, preferably `$()` as backticks can easily be confused with single quotes.

Comment (sschuberth) [reply antenna-documentation/scripts/documentation_release.sh]: But this is probably not needed anyway as you can explicitly push to a fully qualified URL:

    git push https://${GITHUB_USER}:${GITHUB_PASSWORD}@... gh-pages

File: antenna-documentation/scripts/documentation_snapshot.sh [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  git checkout ${CURRENT_BRANCH}
+}
+
+pushChanges() {
+  git push `git config remote.origin.url | sed "s/https:\/\//https:\/\/${GITHUB_USER}:${GITHUB_PASSWORD}@/g"` gh-pages
+}
+
+  createCurrentSnapshotDocumentation
+  createNewIndexRootFile
+  pushChanges
```
Comment (sschuberth) [inline antenna-documentation/scripts/documentation_snapshot.sh]: Indentation seems to be wrong.

**Comment by sschuberth** [comment]
Overall, I believe the approach to use custom scripts instead of the `site-maven-plugin` (aka [github-site-plugin](https://github.com/github/maven-plugins/tree/master/github-site-plugin)) plugin is perfectly feasible and more flexible.

File: antenna-documentation/scripts/createOrUpdateVersionDocumentationFolder.sh (hunk 1 above)
Comment (neubs-bsi) [reply antenna-documentation/scripts/createOrUpdateVersionDocumentationFolder.sh]: no you are right, this is redundant. We just want the documentation of this one branch

**Pull request closed without merging**